cgi-bin/LJ/Talk.pm

```diff
@@ -1505,11 +1546,147 @@ sub talkform {
         return "Cannot load comment information." unless $comment;
     }
 
-    my $pics  = LJ::Talk::get_subjecticons();
-    my $entry = LJ::Entry->new( $journalu, ditemid => $opts->{ditemid} );
+    my $subjecticons = LJ::Talk::get_subjecticons();
+    my $entry        = LJ::Entry->new( $journalu, ditemid => $opts->{ditemid} );
+    my @userpics     = LJ::icons_for_remote($remote);
+
+    my $basesubject = $form->{subject} || "";
+    if ( !$editid && $opts->{replyto} && !$basesubject && $parpost->{'subject'} ) {
+        $basesubject = $parpost->{'subject'};
+        $basesubject =~ s/^Re:\s*//i;
+        $basesubject = "Re: $basesubject";
+    }
+
+    my $screening = LJ::Talk::screening_level( $journalu, $opts->{ditemid} >> 8 ) // '';
+
+    my $default_usertype = '';    # One of anonymous, openid, openid_cookie, cookieuser, user.
+    if ( $form->{usertype} ) {
+
+        # Partial form was submitted; pick up where we left off.
+        $default_usertype = $form->{usertype};
+    }
+    elsif ($remote) {
+
+        # If logged-in user isn't allowed to comment we end up with no default,
+        # which seems correct.
+        if ( $remote->openid_identity ) {
+            $default_usertype = 'openid_cookie';
+        }
+        else {
+            $default_usertype = 'cookieuser';
+        }
+    }
+    elsif ( $journalu->{'opt_whocanreply'} eq "all" ) {
+        $default_usertype = 'anonymous';
+    }
+    else {
+        $default_usertype = 'user';
+    }
+
+    # Variables for talkform.tt
+    my $template_args = {
+        hidden_form_elements => '',
+        form_url             => LJ::create_url( '/talkpost_do', host => $LJ::DOMAIN_WEB ),
+        errors               => $opts->{errors},
+        create_link          => '',
+        subjecticons         => $subjecticons,
+
+        public_entry     => $entry->security eq 'public',
+        default_usertype => $default_usertype,
+
+        comment => {
+            editid      => $editid,
+            editreason  => $comment ? $comment->edit_reason : '',
+            oidurl      => $form->{oidurl},
+            oiddo_login => $form->{oiddo_login},
+            user        => $form->{userpost}
+                || $form->{cookieuser},
+            body        => $form->{body},
+            subject     => $basesubject,
+            subjecticon => $form->{subjecticon}
+                || 'none',    # a subjecticon ID
```

Comment:
This will get sent in the form submission. The reason that's okay is because `LJ::Talk::init()` special-cases both `'none'` and `''` as a blank subjecticon field.